### DISCUSSION TITLE
Discussions

### DISCUSSION THREAD

**Discussion by darvil82** [DarviL82/DarviLStuff#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by DavidGeeraerts** [comment]
Discussions was mentioned during the [2020 Github Universe](https://githubuniverse.com/)!
I'm also curious about giving it a try.
Somewhat off topic, though it does share `color` as a key word:
I wrote this program a long time ago --so there is lots of ugly code--: Windows Command Shell Color Explorer - [CSCE](https://github.com/DavidGeeraerts/CSCE)
If by chance it perks your curiosity, it seems it would be interesting to contemplate incorporating what you've done with echoc into CSCE.

**Reply by darvil82** [reply]
Oh mate, seems like you put a lot of effort doing that! Feel free to do whatever you want with echoc in there! By the way, after giving your script a try, I got kinda impressed about how you get random values for colors... And you used `get-random` from powershell! I didn't know there was a module for doing exactly that, it might be very useful in a future... Who knows... Maybe for an update of echoc? hehehe

**Reply by DavidGeeraerts** [reply]
Yeah, there was a bit of effort. I did that project way back to get better at creating cmd programs. Too bad I didn't have it in git from the beginning, it would be interesting to show you how I dealt with the `%RANDOM%` limitation, which was never really that good, which is why I resorted to leveraging powershell.
From what I remember, the program would generate a random number `%RANDOM%`, assign it to a variable, and then do a check where that random number fell with regards to predefined bucket for a color:
0 1-2000
1 2001-4000
etc.
I took the max number that  `%RANDOM%` generates and divided it by the number of colors, which is 16: `0 1 2 3 4 5 6 7 8 9 A B C D E F`
...and made the buckets. Depending on the timeout, the random number was often in the same bucket --so in some ways it didn't work that well.

One of the most impressive shell programs has got to be [BASHTOP](https://github.com/aristocratos/bashtop) --but that's for *NIX
Wonder if that could be pulled off for Windows command shell with powershell leveraged (?).  

I'm waiting for the right moment to delve deeply into **echoc**, but I think that might be a while.
Right now I'm trying to get the first incarnation of **ADDS_Tool** released/working.
_(If you're into Windows Active Directory, let me know.)_

**Reply by darvil82** [reply]
About the random thing, take a look at [this thing](https://stackoverflow.com/questions/5777400/how-to-use-random-in-batch-script) that I just found, it could be used for this like `SET /A test=%RANDOM% * 16 / 32768` and it would give a value from `0` to `15`, which is exactly what is needed!
Also I unfortunately never used Active Directory...

**Reply by DavidGeeraerts** [reply]
Very cool, thanks for sharing.
Useful so as to not have to bring in powershell.
